packages/start-slicemachine/src/legacyMigrations/migrateAssets.ts

```diff
@@ -0,0 +1,174 @@
+import * as path from "node:path";
+import * as fsSync from "node:fs";
+import { SliceMachineManager } from "@slicemachine/manager";
+import {
+	SharedSliceContent,
+	Document,
+} from "@prismicio/types-internal/lib/content";
+import { DocumentMock, SharedSliceMock } from "@prismicio/mocks";
+
+const MOCKS_FILE_NAME = "mocks.json";
+const MOCK_CONFIG_FILE_NAME = "mock-config.json";
+
+const createPathToDeprecatedLibrary = (cwd: string) =>
+	path.join(cwd, ".slicemachine");
+const createPathToCustomTypesAssets = (cwd: string) =>
+	path.join(createPathToDeprecatedLibrary(cwd), "assets", "customtypes");
+
+const safeUnlink = (pathToUnlink: string, type: "file" | "folder") => {
+	try {
+		if (type === "file") {
+			fsSync.unlinkSync(pathToUnlink);
+		} else {
+			fsSync.rmSync(pathToUnlink, { recursive: true });
+		}
+	} catch (_) {}
+};
+
+const ensureOrGenerateSliceScreenshot = (
+	variationsIDs: string[],
+	targetPathToSliceFolder: string,
+	deprecatedPathToSliceAssets: string,
+) => {
+	variationsIDs.forEach((variationID) => {
+		const targetPathToVariationScreenshot = path.join(
+			targetPathToSliceFolder,
+			`screenshot-${variationID}.png`,
+		);
+		if (!fsSync.existsSync(targetPathToVariationScreenshot)) {
+			const deprecatedPathToVariationScreenshot = path.join(
+				deprecatedPathToSliceAssets,
+				variationID,
+				"preview.png",
+			);
+			if (fsSync.existsSync(deprecatedPathToVariationScreenshot)) {
+				fsSync.renameSync(
+					deprecatedPathToVariationScreenshot,
+					targetPathToVariationScreenshot,
+				);
+			}
+		}
+	});
+};
+
+const ensureOrGenerateMockFile = (
+	targetPathToMocks: string,
+	deprecatedPathToMocks: string,
+	validator: (str: string) => boolean,
+	generate: () => Record<string, unknown>,
+) => {
+	try {
+		const rawMockContent = (() => {
+			if (fsSync.existsSync(targetPathToMocks)) {
+				return fsSync.readFileSync(targetPathToMocks, "utf-8");
+			}
+			if (fsSync.existsSync(deprecatedPathToMocks)) {
+				return fsSync.readFileSync(deprecatedPathToMocks, "utf-8");
+			}
+
+			return null;
+		})();
+
+		const regeneratedMocksString = JSON.stringify(generate(), null, 2);
+		const isValidMock = (() => {
+			try {
+				return rawMockContent !== null && validator(rawMockContent);
+			} catch {
+				return false;
+			}
+		})();
+
+		if (!isValidMock) {
+			fsSync.writeFileSync(targetPathToMocks, regeneratedMocksString);
+		} else if (rawMockContent) {
+			fsSync.writeFileSync(targetPathToMocks, rawMockContent);
+		} else {
+			fsSync.writeFileSync(targetPathToMocks, regeneratedMocksString);
+		}
+	} catch (error) {}
```

Comment:
Same question here about seeing this in Sentry? We don't necessarily need to throw (if we don't want to stop the process), but can manually trigger the log

Reply:
Added in: [39e87f0](https://github.com/prismicio/slice-machine/pull/862/commits/39e87f06999e1fbc1cec298d8db2fe8eec2161c0), let me know!